tests/test_linked_list.py

```diff
@@ -125,3 +125,50 @@ def test_append_with_tail_flag():
     singly_linked_list.append(7, tail=True)
     assert len(singly_linked_list) == 2
     assert singly_linked_list._head.next.value == 7
+
+# Test merge_sort 
+def test_reverse_merge_sort():
+    l = SinglyLinkedList()
+    l.append(6)
+    l.append(1)
+    l.append(2)
+    l.append(4)
+    l.append(8)
+    l.append(9)
+    l.append(5)
+    l.append(7)
+    l.append(3)
+    l = l.sort()
+    assert l._head.value == 1
+    assert l._head.next.value == 2
+    assert l._head.next.next.value == 3
+    assert l._head.next.next.next.value == 4
+    assert l._head.next.next.next.next.value == 5
+    assert l._head.next.next.next.next.next.value == 6
+    assert l._head.next.next.next.next.next.next.value == 7
+    assert l._head.next.next.next.next.next.next.next.value == 8
+    assert l._head.next.next.next.next.next.next.next.next.value == 9
+    assert l._head.next.next.next.next.next.next.next.next.next == None
+
```

Comment:
You can write less code in this test by doing so..
```python
    list_to_compare = [1, 2, 3, 4, 5, 6, 7, 8, 9]

    #has_same_size = len(list_to_compare) == len(l)
    has_equal_values = all(
        map(lambda x: x[0] == x[1], zip(l, list_to_compare))
    )
    assert has_equal_values is True
    #assert has_same_size is True
```

Reply:
Done!